### DISCUSSION TITLE
WSL Insider on stable Windows

### DISCUSSION THREAD

**Discussion by andrey-skat** [microsoft/WSL#6536] [General]
Hi, is it possible to make separate WSL insider previews without the need to install Windows Insider? 
For now, we have only the choice to use unstable windows or use old WSL without bug fixes and useful features.

**Comment by andrey-skat** [comment]
If this is not the right place for this question, please tell me where it better to ask? :)

**Comment by Biswa96** [comment]
What useful feature are you interested in?

**Reply by andrey-skat** [reply]
Every feature since 19041 :) https://github.com/MicrosoftDocs/WSL/releases
There are many bugfixes or features like `wsl.exe --update` and `wsl.exe --cd` that need to wait for almost a year until they land to release

**Reply by Biswa96** [reply]
`wsl.exe --cd` is available since Microsoft devs added wsl option in context menu. For `wsl.exe --update` alternative, you can install Linux kernel msi installer from Microsoft update catalog.

Sometimes, copying wsl.exe (or wslclient.dll or wslapi.dll) from newer insider builds helps to gain new hidden features.

**Reply by andrey-skat** [reply]
It seems not to be right way to get wsl up to date... Office has insider preview, windows terminal, even sticky notes.
Windows releases every 6 months, but WSL actually every 1 year :(

**Comment by romanlex** [comment]
it's a good idea
I don't want to install Windows Insider but want to get all the last updates with WSL
I think it's very important for many peoples who want to migrate all development environments from Linux machine to WSL

**Comment by andrey-skat** [comment]
@craigloewen-msft sorry for the direct mention. Did you consider (and is it possible) to make an independent insider preview for WSL?

**Reply by craigloewen-msft** [reply]
No problem! Thanks for the ping! 

Yes this is something that we've thought of before, and we have seen the request to use new versions of WSL without taking a new Windows image. Right now WSL ships with the Windows image, so currently there isn't an easy way to accomplish this. However, since WSL 2 runs as a virtual machine it makes it much more of a possibility for us to ship independently of the Windows image, making this request more possible in the future. Unfortunately I don't have any more updates or context beyond that, but I wanted to let you know it is something that we have thought about and believe would be valuable for users!

**Comment by andrey-skat** [comment]
Just installed update 21h1 and found that there is no `--update`  wsl command which was added almost a year ago :(
